### PR TITLE
Cleanup register resets

### PR DESCRIPTION
**Pull request by pudelkoM**
This PR moves the register reset annotations behind the `incompatible_enable_register_reset_annotations` flag.

### CODE REVIEW

**Comment by ccascone** [comment]
@Yi-Tseng and I tried https://github.com/stratum/stratum/pull/348/commits/5dd8958df23f2c546fe8f5723e3bb28f44f9eb75 on HW and the register reset didn't seem to work. When generating traffic matching a watchlist rule we didn't see INT reports being generated every second. We then deployed the stratum-bfrt image based on the previous commit https://github.com/stratum/stratum/pull/348/commits/e456e5d23dffd3167957ee263e7041dfd6b803fa and it worked. So we believe that changes introduced in https://github.com/stratum/stratum/pull/348/commits/5dd8958df23f2c546fe8f5723e3bb28f44f9eb75 cause the reset logic to stop working.

**Comment by ccascone** [comment]
I confirm the issue goes away when deploying an image based on this PR. The register reset logic works as expected when passing the `incompatible_enable_register_reset_annotations` flag.

File: stratum/hal/lib/barefoot/bfrt_table_manager.cc [hunk 1]
```diff
@@ -133,69 +70,89 @@ ::util::Status BfrtTableManager::VerifyForwardingPipelineConfig(
...
+  if (intervals_ms.empty()) {
+    return ::util::OkStatus();
+  }
+  std::sort(intervals_ms.begin(), intervals_ms.end());
+  auto last = std::unique(intervals_ms.begin(), intervals_ms.end());
+  intervals_ms.erase(last, intervals_ms.end());
+  if (intervals_ms.size() != 1) {
+    RETURN_ERROR(ERR_INVALID_PARAM)
+        << "Inconsistent register reset intervals are not supported.";
+  }
```
Comment (Yi-Tseng) [inline stratum/hal/lib/barefoot/bfrt_table_manager.cc:110]: This logic makes sense to me.
Was wondering if we can simplify to this:

```cpp
auto last = std::unique(intervals_ms.begin(), intervals_ms.end());
if (last != (intervals_ms.begin() + 1)) {
  //return error
}
```
```

**Review by Yi-Tseng — APPROVED** [review]

File: stratum/hal/lib/barefoot/bfrt_table_manager.cc (hunk 1 above)
Comment (Yi-Tseng) [reply stratum/hal/lib/barefoot/bfrt_table_manager.cc:110]: But we can merge this patch now, I don't really find a big issue.

Comment (pudelkoM) [reply stratum/hal/lib/barefoot/bfrt_table_manager.cc:110]: I really like you idea, but have two concerns:

- I don't want to test this change again
- It makes the code in more unreadable and harder to understand

**Pull request merged**